Xerpa/clean_architecture_ex
# Contract

### What have been done?

Added contract module to be used at contracts within applications.

### How to test

1. Add the library to an existing Elixir project:

```elixir
{:clean_architecture, "~> 0.1", path: "YOUR_PATH/clean_architecture"},
```

2. Create a contract using `CleanArchitecture.Contract`:

```elixir
defmodule YourAPP.Contracts.SomeAction do
  use CleanArchitecture.Contract

  embedded_schema do
    field :id, Ecto.UUID
    field :name, :string
  end

  def changeset(%{} = attrs) do
    %__MODULE__{}
    |> cast(attrs, [:id, :name])
    |> validate_required([:id])
    |> validate_required_if_attribute_is_present(:name, attrs)
  end
end
```

3. Test it to see if runs as expected

```
YourAPP.Contracts.SomeAction.validate_input(%{})
```
